src/FSharp.Core/mailbox.fs

```diff
@@ -331,6 +333,13 @@ type Mailbox<'Msg>(cancellationSupported: bool) =
 
     interface System.IDisposable with
         member _.Dispose() =
+            lock syncRoot (fun () ->
+                if isNotNull inboxStore then
+                    inboxStore.Clear()
+
+                arrivals.Clear()
```

Comment:
What's the view on finishing processing the existing items in the queue VS dropping them ?

Reply:
It seems in the issue the expected behavior that nothing will happen. Undoubtedly, this will be breaking behavior, as well as, for example, an exception for the existing items. Just with an exception, the runtime will immediately report the changed behavior